src/utils/export/CsvExportLink.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useLazyQuery } from '@apollo/client';
+import { IconFileDownload } from '@trussworks/react-uswds';
+import { Parser, transforms } from 'json2csv';
+
+import GetAllModelPlans from 'queries/GetAllModelData';
+import {
+  GetAllModelData as GetAllModelDataType,
+  GetAllModelData_modelPlanCollection as AllModelDataType
+} from 'queries/types/GetAllModelData';
+
+import { csvFields, fieldsToUnwind } from './CsvData';
+
+const csvFormatter = (csvData: AllModelDataType[]) => {
+  try {
+    const transform = [
+      transforms.unwind({ paths: fieldsToUnwind, blankOut: true })
+    ];
+    const parser = new Parser({
+      // csvFields,
+      fields: csvFields,
+      transforms: transform
+    });
+    const csv = parser.parse(csvData);
+    downloadFile(csv);
+  } catch (err) {
+    // TODO: add more robust error handling: display a modal/message if download fails?
+    console.error(err); // eslint-disable-line
+  }
+};
+
+const downloadFile = (data: string) => {
+  const element = document.createElement('a');
+  const file = new Blob([data], {
+    type: 'text/csv'
+  });
+  element.href = URL.createObjectURL(file);
+  element.download = 'modelPlans.csv'; // TODO: make this configurable
+  document.body.appendChild(element);
+  element.click();
+};
+
+export const CsvExportLink = (): React.ReactElement => {
+  const { t } = useTranslation('home');
+
+  const [fetchModelCSVData] = useLazyQuery<GetAllModelDataType>(
+    GetAllModelPlans
+  );
+
+  const fetchData = async () => {
+    const modelPlans = await fetchModelCSVData();
+    csvFormatter(modelPlans?.data?.modelPlanCollection || []);
+  };
+
+  return (
+    <div>
+      <button type="button" onClick={() => fetchData()}>
+        <IconFileDownload />
+        &nbsp;
+        {t('downloadCSV')}
+      </button>
+    </div>
```

Comment:
This csv button should mirror the figma link here - [figma](https://www.figma.com/file/M72DReE73SSXMN48WZfaKN/MINT---New-Model-Process-(Model-Intake)?node-id=1691%3A237666)